app/common/util/src/main/java/io/syndesis/common/util/Labels.java

```diff
@@ -40,6 +42,11 @@ private Labels() {
      * @return      The sanitized string.
      */
     public static String sanitize(String name) {
+
+        if( name.length()<= MAXIMUM_NAME_LENGTH && VALID_LABEL_REGEX.matcher(name).matches() ) {
```

Comment:
maximum length is 63, not 100, according to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

If I understood @iocanel right, we are striving to remove the need to sanitize id and version labels (but validating instead before, so fail fast if they can't be used).

Not sure whether we use labels for something different, so could be that we get rid of method eventually.

Otherwise I'm perfectly fine with this PR, thanks !

Reply:
oops, sorry. that was the name which can be at max 63 chars.